components/UpsellPlusModal.vue

```diff
@@ -0,0 +1,239 @@
+            </li>
+          </ul>
+        </div>
+
+        <!-- Subscription Buttons -->
+      </div>
+    </template>
+    <template #footer>
+      <div class="flex flex-col items-center w-full">
+        <div class="w-full flex items-center gap-2">
+          <UButton
+            v-if="showYearlyPlan"
+            class="w-full"
+            :label="$t('upsell_plus_yearly_button')"
+            block
+            size="xl"
+            color="primary"
+            :ui="{
+              base: 'py-2 laptop:py-3 rounded-2xl cursor-pointer',
+              label: 'font-bold',
+            }"
+            @click="() => handleSubscribe('yearly')"
+          />
+          <UButton
+            v-if="showMonthlyPlan"
+            class="w-full"
+            :label="$t('upsell_plus_monthly_button')"
+            block
+            size="xl"
+            color="primary"
+            :ui="{
+              base: 'py-2 laptop:py-3 rounded-2xl cursor-pointer',
+              label: 'font-bold',
+            }"
+            @click="() => handleSubscribe('monthly')"
+          />
+        </div>
+
+        <UButton
+          class="mt-4"
+          :label="$t('upsell_plus_modal_close_button')"
+          block
+          size="xl"
+          variant="link"
+          :ui="{
+            base: 'rounded-2xl cursor-pointer',
+            label: 'font-bold',
+          }"
+          @click="handleClose"
+        />
+      </div>
+    </template>
+  </UModal>
+</template>
+
+<script setup lang="ts">
+import type { _RouteLocationBase } from 'vue-router'
+import type { UpsellPlusModalProps } from './UpsellPlusModal.props'
+
+const props = withDefaults(defineProps<UpsellPlusModalProps>(), {
+  isLikerPlus: false,
+  likerPlusPeriod: undefined,
+  isProcessingSubscription: false,
+  hasFreeTrial: false,
+  mustCollectPaymentMethod: false,
+  selectedPricingItemIndex: 0,
+  utmCampaign: undefined,
+  utmMedium: undefined,
+  utmSource: undefined,
+})
+
+const emit = defineEmits<{
+  open: []
+  close: []
+  subscribe: [payload: {
+    plan: SubscriptionPlan
+    utmCampaign?: string
+    utmMedium?: string
+    utmSource?: string
+    redirectInfo?: _RouteLocationBase
+  }]
+}>()
+
+const { t: $t } = useI18n()
+const route = useRoute()
+const showYearlyPlan = ref(false)
+const showMonthlyPlan = computed(() => !props.isLikerPlus)
+
+const redirectInfo = computed((): _RouteLocationBase => (
+  {
+    name: route.name,
+    path: route.path,
+    fullPath: route.fullPath,
+    params: route.params,
+    query: {
+      ...route.query,
+      selectedPricingItemIndex: String(props.selectedPricingItemIndex),
+    },
+    hash: route.hash,
+    meta: route.meta,
+    redirectedFrom: route.redirectedFrom,
+  }))
+
+function handleSubscribe(plan: SubscriptionPlan) {
+  emit('subscribe', {
+    plan,
+    utmCampaign: props.utmCampaign,
+    utmMedium: props.utmMedium,
+    utmSource: props.utmSource,
+    redirectInfo: redirectInfo.value,
+  })
+}
```

Comment:
I think we can store this is enough
```suggestion
const getRouteBaseName = useRouteBaseName()

function handleSubscribe(plan: SubscriptionPlan) {
  emit('subscribe', {
    plan,
    utmCampaign: props.utmCampaign,
    utmMedium: props.utmMedium,
    utmSource: props.utmSource,
    redirectRoute:  {
      name: getRouteBaseName(route),
      params: route.param,
      query: route.query,
      hash: route.hash,
    },
  })
}
```